### PR TITLE
PIM-10869: Image upload security fix

### PR DESCRIPTION
**Pull request by LevFlavien**
**Description (for Contributor and Core Developer)**

Story https://akeneo.atlassian.net/browse/PIM-10869  
Tech desgin https://www.notion.so/akeneo/PIM-10869-Image-Upload-Unrestricted-File-Upload-allows-several-vulnerabilities-fa2e09fbfaa940c1abf90ea4b4455318  
EE PR https://github.com/akeneo/pim-enterprise-dev/pull/17858

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [x] Changelog (maintenance bug fixes)

### CODE REVIEW

**Comment by LevFlavien** [comment]
@olivierpontier Yes. It's part of the global test suite that runs in CI.

**Review by pchasle — APPROVED** [review]

File: src/Akeneo/Tool/Bundle/FileStorageBundle/Controller/InternalApi/UploadImageAction.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        return new JsonResponse(
+            [
+                'originalFilename' => $uploadedFile->getClientOriginalName(),
+                'filePath'         => $file->getKey()
+            ]
+        );
+    }
+
+    protected function storeFile(UploadedFile $uploadedFile): FileInfoInterface
```
Comment (mmetayer) [inline src/Akeneo/Tool/Bundle/FileStorageBundle/Controller/InternalApi/UploadImageAction.php:68]: private (the class is final anyway :slightly_smiling_face:  )

File: src/Akeneo/Tool/Bundle/FileStorageBundle/Validator/Constraints/UploadedFileValidator.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+namespace Akeneo\Tool\Bundle\FileStorageBundle\Validator\Constraints;
+
+use Symfony\Component\HttpFoundation\File;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class UploadedFileValidator extends ConstraintValidator
```
Comment (mmetayer) [inline src/Akeneo/Tool/Bundle/FileStorageBundle/Validator/Constraints/UploadedFileValidator.php:13]: would be great to unit test this class

**Review by mmetayer — APPROVED** [review]
OK for enrichment part :ok_hand: but it would be really great to unit test the new validator

**Review by RiffVentura — APPROVED** [review]

**Pull request merged**